packages/typescript/package.json

```diff
@@ -29,18 +29,18 @@
     "@typescript-eslint/parser": "^5.42.1",
     "eslint": "^8.27.0",
     "eslint-config-prettier": "^8.5.0",
-    "eslint-plugin-import": "^2.26.0",
-    "eslint-plugin-jsdoc": "^39.6.2",
+    "eslint-plugin-import": "^2.27.5",
+    "eslint-plugin-jsdoc": "^41.1.2",
     "eslint-plugin-prettier": "^4.2.1",
     "prettier": "^2.7.1",
-    "typescript": "~4.8.4"
+    "typescript": "~4.8.4 || ^5.0.0"
   },
   "peerDependencies": {
     "@metamask/eslint-config": "^11.0.0",
     "@typescript-eslint/eslint-plugin": "^5.42.1",
     "@typescript-eslint/parser": "^5.42.1",
     "eslint": "^8.27.0",
-    "typescript": "~4.8.4"
+    "typescript": "~4.8.4 || ^5.0.0"
```

Comment:
We had been intentionally using `~` ranges for TypeScript to prevent it from being used on TypeScript versions that may not be compatible with the plugin we're using. i.e. to prevent this error:

>WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

Without this, it's easy to upgrade and not notice any incompatibility until you run the linter locally and see that message.

Reply:
I agree. It's annoying that `@typescript-eslint` isn't automatically compatible with future versions of TypeScript, but I think it's better for our projects to not get updated to a new version of TypeScript prematurely than it is to have to keep the TypeScript versions manually updated. What do you think about changing this to:

``` suggestion
    "typescript": "~4.8.4 || ~5.0.0"
```